api/src/services/access.ts

```diff
@@ -22,18 +28,30 @@ export class AccessService extends ItemsService {
 		data: Partial<Item>,
 		opts: MutationOptions = {},
 	): Promise<PrimaryKey[]> {
-		const result = await super.updateMany(keys, data, opts);
+		if (opts.onRequireUserIntegrityCheck) {
+			// Updating policy attachments might affect the number of admin/app/api users
+			opts.onRequireUserIntegrityCheck(UserIntegrityCheckFlag.All);
+		}
+
+		const result = await super.updateMany(keys, data, { ...opts, userIntegrityCheckFlags: UserIntegrityCheckFlag.All });
 
-		// Some policy attachments has been updated, clear the permissions cache
+		// Some policy attachments have been updated, clear the permissions cache
 		await clearPermissionsCache();
 
 		return result;
 	}
 
 	override async deleteMany(keys: PrimaryKey[], opts: MutationOptions = {}): Promise<PrimaryKey[]> {
+		// Changes here can affect the number of admin/app/api users
+		if (opts.onRequireUserIntegrityCheck) {
+			opts.onRequireUserIntegrityCheck(UserIntegrityCheckFlag.All);
```

Comment:
Do we really need to check `All` here? Wouldn't remaining admins be sufficient?

```suggestion
			opts.onRequireUserIntegrityCheck(UserIntegrityCheckFlag.RemainingAdmins);
```

Reply:
Sadly not, I thought so too at first, but deleting a policy can move users from app or admin users to api users, which have a separate limit which might then be breached.

Reply:
I seeee ✔️